src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -472,52 +512,181 @@ private void CreateRegionLinks()
                 if (connections.ContainsKey(connectionKey) || connections.ContainsKey(reverseConnectionKey))
                     continue;
 
-                var pathToAdjacent = GetLeastIntersectingPath(region, adjacent);
+                var (pathToAdjacent, startVertical, endVertical) = GetLeastIntersectingPath(region, adjacent);
 
-                connections.Add(connectionKey, pathToAdjacent);
+                connections.Add(connectionKey, (pathToAdjacent, startVertical, endVertical));
             }
         }
 
         AdjustPathEndpoints();
     }
 
+    private int AdjustPriority(int size1, int priority)
+    {
+        // Prioritize connections to the longer sides
+        priority += size1 * size1;
+
+        return priority;
+    }
+
+    private int GetIShapePathCharacteristics(int startRows, int startColumns, int endRows, int endColumns,
+        bool flip = false)
+    {
+        var priority = 5;
+
+        // Flip: false -> # of Columns, Flip: true -> # of Rows
+        priority = AdjustPriority(flip ? startRows : startColumns, priority);
+        priority = AdjustPriority(flip ? endRows : endColumns, priority);
+        return priority;
+    }
+
+    private int GetLShapePathCharacteristics(int startColumns, int endRows)
+    {
+        var priority = 3;
+        priority = AdjustPriority(startColumns, priority);
+        priority = AdjustPriority(endRows, priority);
+        return priority;
+    }
+
+    private (Vector2 Intermediate1, Vector2 Intermediate2, int Priority) GetZShapePathCharacteristics(int startRows,
+        int startColumns, int endRows, int endColumns, Vector2 startCenter, Vector2 endCenter,
+        Vector2 middlePoint, bool isTrueMiddle, bool flip = false)
+    {
+        Vector2 intermediate1;
+        Vector2 intermediate2;
+        int priority = isTrueMiddle ? 2 : 1;
+        if (!flip)
```

Comment:
I'd add a blank line here to give better readability for the if (as the variables above this aren't used in the if condition, I'd logically separate them with a blank line)